### PR TITLE
Improve Snowflake Iceberg catalog docs 

### PR DESCRIPTION
**Pull request by mayankvadariya**
## Description

Document Snowflake Iceberg catalog limitations

## Release notes

(x) This is not user-visible or is docs only, and no release notes are required.
( ) Release notes are required. Please propose a release note for me.
( ) Release notes are required, with the following suggested text:

```markdown
# Section
* Fix some things. ({issue}`issuenumber`)
```

### CODE REVIEW

File: docs/src/main/sphinx/object-storage/metastores.md [hunk 1]
```diff
@@ -580,6 +580,7 @@ The Nessie catalog does not support [view management](sql-view-management) or
 
 ### Snowflake catalog
 
+Snowflake catalog is a read-only catalog. Additionally, Snowflake doesn't support partitioned Iceberg tables.
```
Comment (mosabua) [inline docs/src/main/sphinx/object-storage/metastores.md:583]: We should move the limitation down and explain more .. similar to how the MV and view support limitation is documented in the others.

**Review by mosabua — CHANGES_REQUESTED** [review]

File: docs/src/main/sphinx/object-storage/metastores.md (hunk 1 above)
Comment (mayankvadariya) [reply docs/src/main/sphinx/object-storage/metastores.md]: sure, updated. Please review.

File: docs/src/main/sphinx/object-storage/metastores.md [hunk 2]
```diff
@@ -612,6 +612,11 @@ iceberg.snowflake-catalog.password=secret
 iceberg.snowflake-catalog.database=db
 ```
 
+The Snowflake catalog has a following limitations in Trino due to the [limitation of Iceberg Snowflake catalog implementation](https://docs.snowflake.com/en/user-guide/tables-iceberg-catalog#limitations):
+- The Snowflake catalog is a read-only catalog.
+- Does not support partitioned Iceberg tables.
```
Comment (ebyhr) [inline docs/src/main/sphinx/object-storage/metastores.md:617]: This sentence is confusing for me. Does it mean the catalog doesn't support "reading" partitioned tables?

Comment (mosabua) [reply docs/src/main/sphinx/object-storage/metastores.md:617]: We have to clarify all this more .. for example also .. what does "read-only" mean? Cant create a table? alter a table? Or is this all covered in the linked docs . then we can omit the whole list. I will review tomorrow.

Comment (mayankvadariya) [reply docs/src/main/sphinx/object-storage/metastores.md]: sorry I meant snowflake Iceberg tables do not support partitions and therefor the snowflake Iceberg catalog.

---
clarified that it supports only SELECT statement as documented in the snowflake iceberg catalog link.

File: docs/src/main/sphinx/object-storage/metastores.md [hunk 3]
```diff
@@ -612,6 +612,21 @@ iceberg.snowflake-catalog.password=secret
...
+When using the Snowflake catalog, data management tasks such as creating tables,
+must be performed in Snowflake because using the catalog from external systems
+like Trino only supports `SELECT` queries as read operations.
+
+Additionally, the [Snowflake-created Iceberg
+tables](https://docs.snowflake.com/en/sql-reference/sql/create-iceberg-table-snowflake)
+do not expose partitioning information, which prevents efficient parallel reads
+and therefore can have significant negative performance implications.
+
+The Snowflake catalog does not support [view management](sql-view-management) or
```
Comment (mayankvadariya) [inline docs/src/main/sphinx/object-storage/metastores.md:624]: nit: Snowflake catalog  -> Snowflake Iceberg catalog ?

**Comment by mosabua** [comment]
Discussed this all a bit more with @mayankvadariya and updated together.

File: docs/src/main/sphinx/object-storage/metastores.md [hunk 4]
```diff
@@ -612,6 +612,21 @@ iceberg.snowflake-catalog.password=secret
...
+[materialized view management](sql-materialized-view-management).
+
+Further information is available in the [Snowflake catalog
```
Comment (mayankvadariya) [inline docs/src/main/sphinx/object-storage/metastores.md:627]: ditto

File: docs/src/main/sphinx/object-storage/metastores.md [hunk 5]
```diff
@@ -612,6 +612,21 @@ iceberg.snowflake-catalog.password=secret
 iceberg.snowflake-catalog.database=db
 ```
 
+When using the Snowflake catalog, data management tasks such as creating tables,
```
Comment (mayankvadariya) [inline docs/src/main/sphinx/object-storage/metastores.md:615]: nit: Snowflake catalog  -> Snowflake Iceberg catalog ?

File: docs/src/main/sphinx/object-storage/metastores.md [hunk 6]
```diff
@@ -577,7 +578,6 @@ The Nessie catalog does not support [view management](sql-view-management) or
 [materialized view management](sql-materialized-view-management).
 
 (iceberg-snowflake-catalog)=
-
```
Comment (mayankvadariya) [inline docs/src/main/sphinx/object-storage/metastores.md:580]: revert?

**Review by mayankvadariya — COMMENTED** [review]
Thank you. Looks good to me.

File: docs/src/main/sphinx/object-storage/metastores.md (hunk 6 above)
Comment (mosabua) [reply docs/src/main/sphinx/object-storage/metastores.md:580]: no ... these empty lines are redundant and we want to remove them

File: docs/src/main/sphinx/object-storage/metastores.md (hunk 3 above)
Comment (mosabua) [reply docs/src/main/sphinx/object-storage/metastores.md:624]: I dont think so .. we dont say REST Iceberg catalog, Nessie Iceberg catalog, or JDBC Iceberg catalog either ..  currently its all consistent and relies on the context that they are all Iceberg catalogs (metastores)

Comment (mayankvadariya) [reply docs/src/main/sphinx/object-storage/metastores.md:624]: Thought of it to add more clarification as to Snowflake catalog could have different meaning in Trino context.

Comment (mayankvadariya) [reply docs/src/main/sphinx/object-storage/metastores.md:624]: Thought of it to add more clarification as tSnowflake catalog could have different meaning in Trino context.

**Review by mosabua — APPROVED** [review]
Looks good now.

File: docs/src/main/sphinx/object-storage/metastores.md [hunk 7]
```diff
@@ -612,6 +612,21 @@ iceberg.snowflake-catalog.password=secret
 iceberg.snowflake-catalog.database=db
 ```
 
+When using the Snowflake catalog, data management tasks such as creating tables,
+must be performed in Snowflake because using the catalog from external systems
+like Trino only supports `SELECT` queries as read operations.
```
Comment (ebyhr) [inline docs/src/main/sphinx/object-storage/metastores.md:617]: I believe there are more statements. e.g., DESCRIBE, SHOW
No need to mention SELECT here in my opinion.

Comment (mayankvadariya) [reply docs/src/main/sphinx/object-storage/metastores.md:617]: should it be reverted to read-only then?

**Review by dain — APPROVED** [review]

File: docs/src/main/sphinx/object-storage/metastores.md (hunk 7 above)
Comment (mosabua) [reply docs/src/main/sphinx/object-storage/metastores.md:617]: I rejigged to keep the SELECT mention but also link to the read operations section. This way we hint at more but dont claim for all of them to work like in connectors.

**Pull request merged**